src/Controller/Rest/DossierRestV2Controller.php

```diff
@@ -152,9 +154,16 @@ public function list(int $beneficiaryId, BeneficiaireProvider $beneficiaireProvi
 
             $user = $this->getUser();
             $isBeneficiaire = $user instanceof User ? $user->isBeneficiaire() : false;
-            $dossiers = $beneficiaire->getDossiers($isBeneficiaire);
 
-            return $this->json($dossiers);
+            $folderId = $this->request->query->get('folder');
+
+            if ($folderId) {
+                $folders = $beneficiaire->getDossiers($isBeneficiaire)->filter(fn (Dossier $folder) => $folder->getDossierParent()?->getId() === (int) $folderId);
+            } elseif ('root' === $this->request->query->get('q')) {
+                $folders = $folderManager->getRootFolders($beneficiaire);
```

Comment:
Isn't it a little weird to use a getter in a case and the manager in the other case ? 
Also if you call this using null as $folderId, does it not return rootFolders only ? 
`$folders = $beneficiaire->getDossiers($isBeneficiaire)->filter(fn (Dossier $folder) => $folder->getDossierParent()?->getId() === (int) $folderId);`